runelite-client/src/main/java/net/runelite/client/ui/NavigationButton.java

```diff
@@ -76,6 +77,11 @@
 	 */
 	private Runnable onSelect;
 
+	/**
+	 * Sets the button's icon to the passed in image.
+	 */
+	private Consumer<BufferedImage> iconConsumer;
```

Comment:
If we are going to require using this to change the icon it might not be great to have a setter for `icon` exposed?